### PR TITLE
feat: Add retry logic for bulk imports based on work in the zed CLI

### PR DESCRIPTION
**Pull request by chancesm**
Tests can be added later. I just wanted an initial review of the code first.

### CODE REVIEW

File: v1/retryable_client.go [hunk 1]
```diff
@@ -0,0 +1,231 @@
...
+	"github.com/cenkalti/backoff/v4"
+	"github.com/samber/lo"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+	"strings"
+	"time"
+)
+
+//----------------------------------------
```
Comment (josephschorr) [inline v1/retryable_client.go:17]: remove this comment

File: v1/retryable_client.go [hunk 2]
```diff
@@ -0,0 +1,231 @@
...
+		"SQLSTATE 23505",     // CockroachDB
+		"Error 1062 (23000)", // MySQL
+	}
+	retryableErrorCodes = []string{
+		"retryable error",                          // CockroachDB, PostgreSQL
+		"try restarting transaction", "Error 1205", // MySQL
+	}
+)
+
+type RetryableClient struct {
```
Comment (josephschorr) [inline v1/retryable_client.go:43]: Add doc comments on all exported functions and types

File: v1/retryable_client.go [hunk 3]
```diff
@@ -0,0 +1,231 @@
...
+	case conflict && conflictStrategy == Fail:
+		return fmt.Errorf("duplicate relationships found")
+
+	default:
+
+		return fmt.Errorf("error finalizing write of %d relationships: %w", len(relationships), err)
+	}
+
+	// it was a successful transaction commit without duplicates
+	if resp != nil {
```
Comment (josephschorr) [inline v1/retryable_client.go:109]: remove

**Comment by chancesm** [comment]
I have read the CLA Document and I hereby sign the CLA

File: v1/retryable_client.go (hunk 2 above)
Comment (chancesm) [reply v1/retryable_client.go:52]: I believe I have done this now. If I missed one, let me know.

File: v1/retryable_client.go [hunk 4]
```diff
@@ -0,0 +1,240 @@
...
+		"SQLSTATE 23505",     // CockroachDB
+		"Error 1062 (23000)", // MySQL
+	}
+	retryableErrorCodes = []string{
+		"retryable error",                          // CockroachDB, PostgreSQL
+		"try restarting transaction", "Error 1205", // MySQL
+	}
+)
+
+// RetryableClient represents an open connection to Authzed with
```
Comment (josephschorr) [inline v1/retryable_client.go:47]: SpiceDB

Comment (chancesm) [reply v1/retryable_client.go:47]: I'll fix this, but I was following the pattern set on the doc comments for the other clients. They all say Authzed.

Comment (josephschorr) [reply v1/retryable_client.go:47]: Yeah, we're transitioning those to make it match the fact that this is open source

File: v1/retryable_client.go (hunk 2 above)
Comment (josephschorr) [reply v1/retryable_client.go:52]: missed one:

> // NewRetryableClient initializes a brand new client for interacting
// with Authzed.

Comment (chancesm) [reply v1/retryable_client.go:52]: Done ✅

**Review by josephschorr — APPROVED** [review]
LGTM

**Pull request merged**